src/Datadog.Trace/TraceId.cs

```diff
@@ -0,0 +1,212 @@
+using System;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+
+namespace Datadog.Trace
+{
+    /// <summary>
+    /// Class representing 64 or 128 bit TraceID.
+    /// </summary>
+    public readonly struct TraceId : IEquatable<TraceId>
+    {
+        private readonly bool _is64Bit;
+        private readonly ulong _higher;
+        private readonly string _string;
+
+        private TraceId(ulong higher, ulong lower)
+        {
+            _is64Bit = false;
+            _higher = higher;
+            Lower = lower;
+            _string = $"{_higher:x16}{Lower:x16}";
+        }
+
+        private TraceId(ulong lower)
+        {
+            _is64Bit = true;
+            _higher = 0;
+            Lower = lower;
+            _string = Lower.ToString(CultureInfo.InvariantCulture);
```

Comment:
This is great, thank you for accepting and incorporating my feedback 😄

Reply:
Thinking more on this logic, I think it's rather strange for me to be asking you to change the default behavior for these lower values. Perhaps we can pass in a `bool` or a `Func` to determine if the string should be represented as decimal or hex so we don't have a built-in assumption that 64-bit id == decimal. It's a strange and not intuitive assumption to have in the code base. What do you think?